services/QuillLMS/spec/controllers/api/v1/activity_sessions_controller_spec.rb

```diff
@@ -74,23 +75,29 @@
         results
       end
 
-      before { put :update, params: { id: activity_session.uid, concept_results: concept_results }, as: :json }
-
       it 'succeeds' do
+        put :update, params: { id: activity_session.uid, concept_results: concept_results }, as: :json
         expect(response.status).to eq(200)
       end
 
       it 'stores the concept results' do
-        activity_session.reload
-        expect(activity_session.old_concept_results.size).to eq 7
+        # Run Sidekiq jobs immediately instead of queuing them
+        Sidekiq::Testing.inline! do
+          expect do
+            put :update, params: { id: activity_session.uid, concept_results: concept_results }, as: :json
+          end.to change { activity_session.reload.old_concept_results.size }.by(3)
+             .and change { activity_session.reload.concept_results.size }.by(3)
+        end
```

Comment:
The original version of those code with `expect ... to eq 7` was counting the items that were created as part of setup as well as the ones created by the function call.  This cleans things up slightly to only count items new from the call.